source/screens/caseScreens/CaseSummary.tsx

```diff
@@ -310,12 +245,77 @@ const CaseSummary = (props) => {
   } = workflow as Workflow;
 
   const [isModalVisible, toggleModal] = useModal();
-  const [isCalculationDetailsVisible, setCalculationDetailsVisibility] =
-    useState(false);
+  const [showRemoveCaseModal, toggleRemoveCaseModal] = useModal();
+
   const decisions = decision?.decisions?.decision
     ? convertDataToArray(decision.decisions.decision)
     : [];
 
+  const canRemoveCase = [ACTIVE_SIGNATURE_PENDING].includes(
+    caseData.status.type
+  );
+
+  const removeCase = async () => {
+    setRemoveCaseState(RemoveCaseState.Loading);
+
+    const result = await remove(`cases/${caseId}`, undefined, undefined);
+
+    if (!result.data.data.code) {
+      deleteCase(caseId);
+
+      setTimeout(() => {
+        navigation.reset({
+          index: 0,
+          routes: [{ name: "App" }],
+        });
+      }, SCREEN_TRANSITION_DELAY);
+    } else {
+      setRemoveCaseState(RemoveCaseState.Error);
+    }
+  };
+
+  const getModalButtonsSet = (
```

Comment:
Only a set for a single button? Should not be plural: `getModalButtonsSet` -> `getModalButtonSet`.